src/background/telemetry.js

```diff
@@ -2,121 +2,189 @@
-        extra_keys: [],
-      },
-    });
+    browser.telemetry.registerEvents(TELEMETRY_CATEGORY, events);
   } catch (e) {
     if (e.message === "Attempt to register event that is already registered.") {
       // eslint-disable-next-line no-console
-      console.log("telemetry events already registered; skipping registration");
+      console.log("Telemetry events already registered; skipping registration.");
     } else {
       throw e;
     }
   }
+}
 
+function registerScalars() {
+  const scalars = {
+    // Current login count, set when the user opens the doorhanger or manage page.
+    "loginCount": {
+      kind: "count",
+      keyed: false,
+      record_on_release: true,
+    },
+
+    // not implemented yet
+
+    // Total number of logins copied or revealed during the current session
+    // TODO: increment a counter with each copy and each reveal
+    "loginsAccessed": {
+      kind: "count",
+      keyed: false,
+      record_on_release: true,
+    },
+  };
   try {
-    browser.telemetry.registerScalars(TELEMETRY_CATEGORY, {
-      "datastoreCount": {
-        kind: "count",
-        keyed: false,
-        record_on_release: false,
-        expired: false,
-      },
-    });
+    browser.telemetry.registerScalars(TELEMETRY_CATEGORY, scalars);
   } catch (e) {
     if (e.message === "Attempt to register scalar that is already registered.") {
       // eslint-disable-next-line no-console
-      console.log("telemetry scalar already registered; skipping registration");
+      console.log("Telemetry scalars already registered; skipping registration.");
     } else {
       throw e;
     }
   }
 }
 
-const exported = {
-  recordEvent(method, object, extra) {
-    return browser.telemetry.recordEvent(TELEMETRY_CATEGORY, method, object, null, extra || null);
-  },
-  scalarSet(name, value) {
-    return browser.telemetry.scalarSet(`${TELEMETRY_CATEGORY}.${name}`, value);
-  },
-};
+async function recordEvent({ method, object, extra = null, value = null }) {
+  let result;
+  try {
+    // Work around bug 1536877: use an embedded API experiment to submit
+    // telemetry data for Firefox less than 68. This can be removed when
+    // our users are at or past Firefox 68.
+    if (await browser.experiments.telemetry.shouldUseEmbedded()) {
+      result = browser.experiments.telemetry.recordEvent(
+        TELEMETRY_CATEGORY, method, object, value, extra);
+    } else {
+      result = browser.telemetry.recordEvent(
+        TELEMETRY_CATEGORY, method, object, value, extra);
+    }
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log(`recordEvent threw on method ${method}, object ${object}, value ${value}, extra ${extra}.`);
+  }
+  return result;
+}
+
+function scalarSet({ name, value }) {
+  return browser.telemetry.scalarSet(`${TELEMETRY_CATEGORY}.${name}`, value);
+}
+
+export function initializeTelemetry() {
+  registerEvents();
+  registerScalars();
+
+  // Record the "startup" event at init time.
```

Comment:
does this fire on browser start even if the user doesn't open the manager? its OK if so I just want to be clear on it before I start using the data

Reply:
Yeah, the addons are loaded at browser startup time